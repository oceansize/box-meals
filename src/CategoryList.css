.Categorylist {
  border: solid 2px red;
}
