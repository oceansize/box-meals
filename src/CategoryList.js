import React, { Component } from 'react';
import './CategoryList.css'
import Category from './Category';
import { dataFetch } from './api/data-fetch';

class CategoryList extends Component {

  constructor (props) {
    super(props);
    this.state = { categories: [] };
  };

  componentWillMount () {
    let endpoint = "categories"
    dataFetch(endpoint)
      .then(items => this.setState({ categories : items}))
  };

  renderCategories () {
    return this.state.categories.map(category => (
      <Category key={category.id} name={category.title} />
    ));
  };

  render() {
    return (
      <ul className="category-list">
        {this.renderCategories()}
      </ul>
    );
  }
}

export default CategoryList;

