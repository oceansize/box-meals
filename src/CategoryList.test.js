import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CategoryList from './Categorylist';
const getData = require('./api/data-fetch');
jest.mock('./api/request');

describe(CategoryList, () => {

  it('matches the component snapshot', () => {
    const tree = renderer.create(
      <CategoryList />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should load categories from API', () => {
    return getData.dataFetch("categories")
    .then(items => {
      expect(items).toBeDefined()
      expect(items[0].title).toEqual('Mocked Drinks Cabinet')
    })
  })
});

