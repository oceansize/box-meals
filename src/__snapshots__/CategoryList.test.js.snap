// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class CategoryList extends /* istanbul ignore next */_react.Component {

  constructor(props) {/* istanbul ignore next */++cov_3bdxopdgj.f[0];++cov_3bdxopdgj.s[0];
    super(props); /* istanbul ignore next */++cov_3bdxopdgj.s[1];
    this.state = { categories: [] };
  }

  componentWillMount() {/* istanbul ignore next */++cov_3bdxopdgj.f[1];
    let endpoint = /* istanbul ignore next */(++cov_3bdxopdgj.s[2], "categories"); /* istanbul ignore next */++cov_3bdxopdgj.s[3];
    /* istanbul ignore next */(0, _dataFetch.dataFetch)(endpoint).
    then(items => {/* istanbul ignore next */++cov_3bdxopdgj.f[2];++cov_3bdxopdgj.s[4];return this.setState({ categories: items });});
  }

  renderCategories() {/* istanbul ignore next */++cov_3bdxopdgj.f[3];++cov_3bdxopdgj.s[5];
    return this.state.categories.map(category =>
    {/* istanbul ignore next */++cov_3bdxopdgj.f[4];++cov_3bdxopdgj.s[6];return (/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Category2.default, /* istanbul ignore next */{ key: category.id, name: category.title, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }));});

  }

  render() {/* istanbul ignore next */++cov_3bdxopdgj.f[5];++cov_3bdxopdgj.s[7];
    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'ul', /* istanbul ignore next */{ className: 'category-list', __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this },
        this.renderCategories()));


  }} matches the component snapshot 1`] = `
<ul
  className="category-list"
/>
`;

exports[`class CategoryList extends _react.Component {

  constructor(props) {
    super(props);
    this.state = { categories: [] };
  }

  componentWillMount() {
    let endpoint = "categories";
    (0, _dataFetch.dataFetch)(endpoint).
    then(items => this.setState({ categories: items }));
  }

  renderCategories() {
    return this.state.categories.map(category =>
    _react2.default.createElement(_Category2.default, { key: category.id, name: category.title, __source: { fileName: _jsxFileName, lineNumber: 21 }, __self: this }));

  }

  render() {
    return (
      _react2.default.createElement('ul', { className: 'category-list', __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this },
        this.renderCategories()));


  }} matches the component snapshot 1`] = `
<ul
  className="category-list"
/>
`;
